### PR TITLE
Feature/merchant

### PR DESCRIPTION
**Pull request by Justnuked**
Merchant feature. Quick coding, will probably require refactor down the line.

### CODE REVIEW

File: behind-the-tankard/Assets/Scripts/AI/INPC.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Assets.Scripts.AI
+{
+    interface INPC
+    {
+
```
Comment (YWAMWeessies) [inline behind-the-tankard/Assets/Scripts/AI/INPC.cs]: is this interface necessary?

Comment (TJ-Smits) [reply behind-the-tankard/Assets/Scripts/AI/INPC.cs]: Good question Yuri, is it?

File: behind-the-tankard/Assets/Scripts/AI/MerchantSpawner.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+using UnityEngine;
+
+public class MerchantSpawner : MonoBehaviour
+{
+
+    public GameObject merchantPrefab;
+
+    private List<Vector3> _spawnPoints = new List<Vector3>();
+    private WorldController _worldController;
+    private bool isSpawned = false;
```
Comment (TJ-Smits) [inline behind-the-tankard/Assets/Scripts/AI/MerchantSpawner.cs]: Misses underscore

File: behind-the-tankard/Assets/Scripts/AI/MerchantSpawner.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if(_worldController.dayController.GetCurrentDayStage() == 0 && !isSpawned)
+        {
+            Vector3 randomSpawnPoint = _spawnPoints[Random.Range(0, _spawnPoints.Count)];
+            isSpawned = true;
+            _merchant = Instantiate(merchantPrefab, randomSpawnPoint, Quaternion.identity);
```
Comment (TJ-Smits) [inline behind-the-tankard/Assets/Scripts/AI/MerchantSpawner.cs]: Can be changed into an object pooler

File: behind-the-tankard/Assets/Scripts/Interactable/Categories/MerchantItemHolder.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+{
+    private Inventory _inventory = new Inventory();
+
+    public int amountBrought { get; set; }
+    public GameObject ItemToDispense { get; set; }
+    public Item itemObj { get; set; }
+
+    public float yOffset;
+
+
```
Comment (TJ-Smits) [inline behind-the-tankard/Assets/Scripts/Interactable/Categories/MerchantItemHolder.cs]: Unnecessary space

File: behind-the-tankard/Assets/Scripts/Interactable/Categories/MerchantItemHolder.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            float cost = itemObj.buyPrice;
+
+            if ((playerInventory.GetMoney() - cost) >= 0f)
+            {
+                playerInventory.RemoveMoney(cost);
+                _inventory.ExchangeItem(item, player);
+            }
+            else
+            {
+                Debug.Log("No money");
```
Comment (TJ-Smits) [inline behind-the-tankard/Assets/Scripts/Interactable/Categories/MerchantItemHolder.cs]: Eventually replaced by a "no"/"failure" sound

File: behind-the-tankard/Assets/Scripts/Interactable/PlayerMovement.cs [hunk 6]
```diff
@@ -75,16 +75,25 @@ private void CheckInventory()
     {
         if(_inventory.HasInventory())
         {
-            NourishmentStats.Types heldItem = _inventory.Items[0].GetComponent<Nourishment>().GetKind();
+            Nourishment heldItem; 
 
-            switch(heldItem)
+            if(_inventory.Items[0].TryGetComponent(out heldItem))
```
Comment (TJ-Smits) [inline behind-the-tankard/Assets/Scripts/Interactable/PlayerMovement.cs]: Thanks for this improvement

**Pull request merged**